syntaxes/go.sum.tmGrammar.json

```diff
@@ -0,0 +1,44 @@
+{
+	"scopeName": "go.sum",
+	"patterns": [
+		{
+			"include": "#checksum"
+		},
+		{
+			"include": "#semver"
+		},
+		{
+			"include": "#unquoted_string"
+		}
+	],
+	"repository": {
+		"checksum": {
+			"comment": "Checksum",
+			"match": "h1:([^\\s]+)=",
+			"captures": {
+				"1": {
+					"patterns": [
+						{
+							"match": "[a-zA-Z\\d+\\/]{43}",
+							"name": "string.unquoted.go.sum"
+						},
+						{
+							"match": ".*",
+							"name": "invalid.illegal.unknown-hash.go.sum"
+						}
+					]
+				}
+			}
+		},
+		"semver": {
+			"comment": "Semver version strings (v1.2.3)",
+			"match": "v(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)(?:-[\\da-z-]+(?:\\.[\\da-z-]+)*)?(?:\\+[\\da-z-]+(?:\\.[\\da-z-]+)*)?",
```

Comment:
the "[official](https://github.com/golang/go/blob/df557fe3ea94ff8888abd6a6ab4cc5d5351d77a6/src/cmd/go/internal/modfetch/pseudo.go#L89)" regexp is: `^v[0-9]+\.(0\.0-|\d+\.\d+-([^+]*\.)?0\.)\d{14}-[A-Za-z0-9]+(\+incompatible)?$`

Reply:
I think that's for the special pseudo version: https://tip.golang.org/cmd/go/#hdr-Pseudo_versions

Reply:
I think I agree about using that regex instead - I wrote one by hand on the Atom repo, but we may want to just use the same regex as the one the language uses.

Reply:
The issue is it only matches pseudo versions: 

```v0.0.0-20161113214103-89cd22812c4f```

It won't match 

```v1.2.3```

At least that's how I read the regex. Is there another one that matches both?

Reply:
I think you are right - the code they use to validate the semver is [here](https://github.com/golang/go/blob/aab0b704d88b7dad10c767ad941619d59887203c/src/cmd/go/internal/semver/semver.go#L150). 

In testing out your branch locally I didn't experience any problems.